payments_service/braintree/forms.py

```diff
@@ -34,6 +46,74 @@ def clean(self):
             raise forms.ValidationError(
                 'Either pay_method_nonce or pay_method_uri can be submitted')
 
+        plan_id = cleaned_data.get('plan_id')
```

Comment:
This reads ok but maybe could add a few comments directly just to explain the logic of the various checks wrt email + donation, user requirements for normal product sub etc.